### PR TITLE
Payments #72

### PR DESCRIPTION
**Pull request by eric-silva-61**
### What this PR does (required):
- Form with name, address, city, state and postal code
- Formik validation for non Stripe form items
- Stripe card number, expiry date and CVC
- Integrated into Profile Settings

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- From profile settings navigate to Payment via the left tab

### Any issues with the current functionality (optional):
- None

### CODE REVIEW

File: client/src/components/PaymentForm/PaymentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        state: Yup.string().required('State is required'),
+        postal_code: Yup.string().required('Postal code is required'),
+      })}
+      onSubmit={handleSubmit}
+    >
+      {({ handleSubmit, handleChange, values, touched, errors, isSubmitting }) => (
+        <form onSubmit={handleSubmit} className={classes.form}>
+          <Grid container spacing={2}>
+            <Grid item xs={12}>
+              <FormLabel htmlFor="name">Name</FormLabel>
```
Comment (rajivtitus) [inline client/src/components/PaymentForm/PaymentForm.tsx:102]: 👍🏽

File: client/src/components/PaymentForm/PaymentForm.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            </Grid>
+            <Grid item xs={12} md={3}>
+              <FormLabel htmlFor="name">State</FormLabel>
+              <Select
+                id="state"
+                name="state"
+                variant="outlined"
+                fullWidth
+                margin="none"
+                //helperText={touched.state ? errors.state : ''}
```
Comment (rajivtitus) [inline client/src/components/PaymentForm/PaymentForm.tsx]: Please remove/uncomment

File: client/src/pages/ProfileSetting/PaymentDetails/PaymentDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { FC } from 'react';
+import { Elements } from '@stripe/react-stripe-js';
+import { loadStripe } from '@stripe/stripe-js';
+import PaymentForm from '../../../components/PaymentForm/PaymentForm';
+import PageContainer from '../../../components/PageContainer/PageContainer';
+
+const PaymentDetails: FC = (): JSX.Element => {
+  const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_PUBLIC_KEY as string);
```
Comment (rajivtitus) [inline client/src/pages/ProfileSetting/PaymentDetails/PaymentDetails.tsx]: I can't remember if you've asked me a question about this 😄  but it's a good practice to load stripe outside of a component's render. Please see image below:

Capture

**Review by rajivtitus — APPROVED** [review]
LGTM! Just a couple of comments, please have a look and address. Thanks

File: client/src/components/PaymentForm/PaymentForm.tsx (hunk 2 above)
Comment (eric-silva-61) [reply client/src/components/PaymentForm/PaymentForm.tsx]: removed

File: client/src/pages/ProfileSetting/PaymentDetails/PaymentDetails.tsx (hunk 3 above)
Comment (eric-silva-61) [reply client/src/pages/ProfileSetting/PaymentDetails/PaymentDetails.tsx]: You mentioned it and I actually read that and intended to do it, then didn't...
Done

**Pull request merged**